### ISSUE TITLE
Damged PDF with some png images

### ISSUE THREAD

**Issue by lugaru90** [dompdf/dompdf#1064]
Hi @ all,

Dompdf Version 0.6.1.

Setting an image with the following command:

$html.='<img src="images/test.png"></img>';

With some PNG Images the output is just fine. But with the most PNG Images, there is an Error on the PDF Readers (Chrome internal and Adobe). With Foxit Reader and Edge i can open the Files just fine.

Any thoughts on this?

Thanks

**Comment by bsweeney** [comment]
So the PDF renders, but does not display correctly? First thought is that it could be some PHP notices related to PNG processing finding their way into the PDF source. Otherwise ... it would help to see either a sample of the PDF or the image causing problems.

**Comment by lugaru90** [comment]
Here is one of the PDF files. The strange thing: i can open it with Chrome without Problems. But it is not working over my PHP Script (which is working fine for other Documents. But maybe my Code also Helps:
```php
<?php
	if(isset($id)){
		$filename = dirname(__FILE__)."/../../pdf/wohnungsBestaetigung/wohnungsBestaetigung-".$id.".pdf"; 
		header("Content-Type: application/pdf"); 
		header("Content-Length: ".filesize($filename)); 
		header('Content-Transfer-Encoding: binary');
		header('Accept-Ranges: bytes');
		header('Content-Disposition: inline; filename="wohnungsBestaetigung-'.$id.'.pdf"'); 
		readfile($filename); 
	}
?>
```
[wohnungsBestaetigung-6573.pdf](https://github.com/dompdf/dompdf/files/49862/wohnungsBestaetigung-6573.pdf)

**Comment by bsweeney** [comment]
Ah, so you generate the PDF and save it on the server, then later you access the PDF via another script which returns it using the above code? What kind of error is given by the readers?

I would start by removing two headers `Accept-Range` and `Content-Transfer-Encoding`.

I'm not really sure how web servers deal with the `Accept-Range` header when serving generated content. dompdf will supply this header, but only conditionally if indicated in the options. I really need to research this a bit more to better understand the interaction between PHP and the web server if this header is present.

As for `Content-Transfer-Encoding`, the actual HTTP header is `Transfer-Encoding` and I'm just not familiar enough with this to even make a guess as to how it should be used.

At any rate, there's nothing written into the script to handle transfer encoding or accept ranges, so I'd leave it to the web server to figure out.

**Comment by lugaru90** [comment]
Yes that is correct. The Reader just states Error on loading the Document.

Already tried to remove this statements. Doesn't change anything -.-

The Strange thing is, that it works with some images and some doesn't work. I create a png Barcode for Bills. If i copy over this generated png and give it the name test.png it renders fine. If i use a test image from the net or my png file / image it dosn't work.

**Comment by bsweeney** [comment]
Is the PDF you uploaded one that was giving an error? For one of the PDFs that's giving you problems can you change the content disposition to attachment, save the PDF, then uploaded it here so I can take a look at it? Sometimes an application that *can* read the PDF will correct any problems it found when saving.

**Comment by lugaru90** [comment]
Hey,

yes it is one of the documents giving an error. I just downloaded this file from the Server over WinSCP. So the Error should stil exist.
